From 4c3fad710c5bc702e28ae2bcc3c73b0547c4954d Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Wed, 19 Nov 2025 14:19:36 +0200
Subject: [PATCH 1/2] ethosu: drop file names from the generated file

Having file names and dates in the generated file affects
reproducibility. Build systems (like OE) error out on the gen_header.py
output, because it can contain full paths. Drop file list from the
generated file.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Backport [https://gitlab.freedesktop.org/mesa/mesa/-/commit/cdb6468c53ef486f237f3d3276013febbbabba14]
---
 src/gallium/drivers/ethosu/gen_header.py | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/gallium/drivers/ethosu/gen_header.py b/src/gallium/drivers/ethosu/gen_header.py
index b54516a812c7..44471893594b 100644
--- a/src/gallium/drivers/ethosu/gen_header.py
+++ b/src/gallium/drivers/ethosu/gen_header.py
@@ -29,21 +29,9 @@ def dump_c(args, guard, func):
 This file was generated by the rules-ng-ng gen_header.py tool in this git repository:
 http://gitlab.freedesktop.org/mesa/mesa/
 git clone https://gitlab.freedesktop.org/mesa/mesa.git
-
-The rules-ng-ng source files this header was generated from are:
 """)
-	maxlen = 0
-	for filepath in p.xml_files:
-		maxlen = max(maxlen, len(filepath))
-	for filepath in p.xml_files:
-		pad = " " * (maxlen - len(filepath))
-		filesize = str(os.path.getsize(filepath))
-		filesize = " " * (7 - len(filesize)) + filesize
-		filetime = time.ctime(os.path.getmtime(filepath))
-		print("- " + filepath + pad + " (" + filesize + " bytes, from " + filetime + ")")
 	if p.copyright_year:
 		current_year = str(datetime.date.today().year)
-		print()
 		print("Copyright (C) %s-%s by the following authors:" % (p.copyright_year, current_year))
 		for author in p.authors:
 			print("- " + author)
-- 
2.51.0

